test/e2e-test/component_version_test.go

```diff
@@ -297,6 +303,182 @@ var _ = Describe("Versioning mechanism of components", func() {
+
+			//Clean
+			k8sClient.Delete(ctx, &appconfig)
+			k8sClient.Delete(ctx, &comp1)
+			k8sClient.Delete(ctx, &comp2)
+		})
+	})
+
+	When("Components have componentName and without revision-enabled trait", func() {
+		It("should create workloads with name of component and replace the old revision", func() {
+
+			By("Create trait definition")
+			var td v1alpha2.TraitDefinition
+			Expect(readYaml("testdata/revision/trait-def-no-revision.yaml", &td)).Should(BeNil())
+			var gtd v1alpha2.TraitDefinition
+			if err := k8sClient.Get(ctx, client.ObjectKey{Name: td.Name, Namespace: td.Namespace}, &gtd); err != nil {
+				Expect(k8sClient.Create(ctx, &td)).Should(Succeed())
+			} else {
+				td.ResourceVersion = gtd.ResourceVersion
+				Expect(k8sClient.Update(ctx, &td)).Should(Succeed())
+			}
+
+			By("Create Component v1")
+			var comp1 v1alpha2.Component
+			Expect(readYaml("testdata/revision/comp-v1.yaml", &comp1)).Should(BeNil())
+			Expect(k8sClient.Create(ctx, &comp1)).Should(Succeed())
+
+			By("Create AppConfig with component")
+			var appconfig v1alpha2.ApplicationConfiguration
+			Expect(readYaml("testdata/revision/app.yaml", &appconfig)).Should(BeNil())
+			Expect(k8sClient.Create(ctx, &appconfig)).Should(Succeed())
+
+			By("Workload created with component name")
+			var w1 unstructured.Unstructured
+			Eventually(
+				func() error {
+					w1.SetAPIVersion("example.com/v1")
+					w1.SetKind("Bar")
+					return k8sClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: componentName}, &w1)
+				},
+				time.Second*15, time.Millisecond*500).Should(BeNil())
+
+			k1, _, _ := unstructured.NestedString(w1.Object, "spec", "key")
+			Expect(k1).Should(BeEquivalentTo("v1"), fmt.Sprintf("%v", w1.Object))
+
+			By("Create Component v2")
+			var comp2 v1alpha2.Component
+			Expect(readYaml("testdata/revision/comp-v2.yaml", &comp2)).Should(BeNil())
+			k8sClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: componentName}, &comp1)
+			comp2.ResourceVersion = comp1.ResourceVersion
+			Expect(k8sClient.Update(ctx, &comp2)).Should(Succeed())
+
+			By("Workload exist with revisionName v2")
+			var w2 unstructured.Unstructured
+			Eventually(
+				func() string {
+					w2.SetAPIVersion("example.com/v1")
+					w2.SetKind("Bar")
+					err := k8sClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: componentName}, &w2)
+					if err != nil {
+						return ""
+					}
+					k2, _, _ := unstructured.NestedString(w2.Object, "spec", "key")
+					return k2
+				},
+				time.Second*15, time.Millisecond*500).Should(BeEquivalentTo("v2"))
+
+			By("Check AppConfig status")
+			Eventually(
+				func() error {
+					return k8sClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: appconfig.Name}, &appconfig)
+				},
+				time.Second*15, time.Millisecond*500).Should(BeNil())
+
+			Expect(len(appconfig.Status.Workloads)).Should(BeEquivalentTo(1))
```

Comment:
Why not check that its revision name should be v2 too

Reply:
sure

Reply:
I have checked it in line 459